compiler/src/dmd/lexer.d

```diff
@@ -2582,8 +2582,15 @@ class Lexer
         {
             /* C11 6.4.4.2 doesn't actually care if it is not representable if it is not hex
              */
-            const char* suffix = (result == TOK.float32Literal || result == TOK.imaginary32Literal) ? "f" : "";
-            error(scanloc, "number `%s%s` is not representable", sbufptr, suffix);
+            const char* suffix = [TOK.float32Literal: "f".ptr,
+                                TOK.float64Literal: "".ptr,
+                                TOK.float80Literal: "L".ptr][result];
+            const char* type = [TOK.float32Literal: "`float`".ptr,
+                                TOK.float64Literal: "`double`".ptr,
```

Comment:
Instead of creating yet another copy of the "float" string, etc., use Token.toChars(TOK.float32Literal).

Reply:
There is no copy created - dmd, gdc, and ldc as far as I've ever seen all merge string literals in the resulting binary

Reply:
These also give the wrong string with toChars() AFAICT